tools/c7n_azure/c7n_azure/storage_utils.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n.utils import local_session
+from c7n_azure.session import Session
+from c7n_azure.utils import ResourceIdParser
+
+from azure.storage.queue import QueueService
+from azure.storage.blob import BlockBlobService
+from six.moves.urllib.parse import urlparse
+
+try:
+    from functools import lru_cache
+except ImportError:
+    from backports.functools_lru_cache import lru_cache
+
+
+class StorageUtilities(object):
+
+    @staticmethod
+    @lru_cache()
+    def get_storage_client_by_uri(storage_uri):
+        parts = urlparse(storage_uri)
+        storage_name = str(parts.netloc).partition('.')[0]
+        container_name = parts.path.partition('/')[2]
+
+        account = StorageUtilities.get_storage_account_by_name(storage_name)
+        key = StorageUtilities.get_storage_keys(account.id)[0].value
+
+        blob_service = BlockBlobService(account_name=storage_name, account_key=key)
+        blob_service.create_container(container_name)
+
+        return blob_service, container_name
+
+    @staticmethod
+    @lru_cache()
+    def get_queue_client_by_uri(queue_uri):
+        parts = urlparse(queue_uri)
```

Comment:
There is a lot of duplicate code for these 2 methods, we also might need the same methods for tables and files clients later, maybe consider reusing?

Reply:
Good point, I still haven't recovered from lack of sleep at PyCon :)   Will make that nicer.